src/workerd/jsg/jsvalue.h

```diff
@@ -427,14 +427,15 @@ inline kj::Maybe<T&> JsValue::tryGetExternal(Lock& js, const JsValue& value) {
 
 template <typename T>
 inline kj::Array<T> JsString::toArray(Lock& js, WriteOptions options) const {
+  v8::String::ValueView view(js.v8Isolate, inner);
   if constexpr (kj::isSameType<T, kj::byte>()) {
-    KJ_ASSERT(inner->ContainsOnlyOneByte());
-    auto buf = kj::heapArray<kj::byte>(inner->Length());
-    inner->WriteOneByte(js.v8Isolate, buf.begin(), 0, buf.size(), options);
+    KJ_ASSERT(view.is_one_byte());
```

Comment:
I think this behaves differently if the V8 string is backed by two-byte characters but all the characters happen to be in the 0-255 range.

Reply:
Yes, you're right. I think that's why `ContainsOnlyOneByte()` scans the whole string in order to understand even if the backing store is two-byte characters, all inputs are one-byte.